core/src/main/java/com/linecorp/armeria/server/ServerHttp1ObjectEncoder.java

```diff
@@ -47,6 +50,7 @@
     private final boolean enableServerHeader;
     private final boolean enableDateHeader;
     private final Http1HeaderNaming http1HeaderNaming;
+    private final WebSocketUpgradeContext webSocketUpgradeContext = new WebSocketUpgradeContext();
```

Comment:
Could we activate WebSocket protocol only when `WebSocketService`s are added?

Reply:
That's a good suggestion. Let me add NoopCotext.